src/controllers/helpers/pagination.rs

```diff
@@ -73,7 +73,7 @@ impl PaginationOptionsBuilder {
         self
     }
 
-    pub(crate) fn gather(self, req: &mut dyn RequestExt) -> AppResult<PaginationOptions> {
```

Comment:
@Turbo87 I don't know if I dared too much here, I haven't seen a real motivation for that mutable reference, and in facts the rest of the code is untouched

Reply:
this seems like a useful change, but it would be best if such refactorings were kept out of this PR for a functional change. (see chapter 6 of https://mtlynch.io/code-review-love/ for a much better explanation :D)